pipelines/auth/pip.yaml

```diff
@@ -0,0 +1,53 @@
+name: Configure pip to fetch Chainguard Libraries dependencies.
+
+needs:
+  packages:
+    - busybox
+    - curl
+    - jq
+
+inputs:
+  identity:
+    description: the identity to use for Chainguard Libraries
+    default: "720909c9f5279097d847ad02a2f24ba8f59de36a/a49c7fedc33adf69"
+  group:
+    description: the group to use for Chainguard Libraries
+    default: "720909c9f5279097d847ad02a2f24ba8f59de36a"
+  guarded:
+    description: do not fallback to other Python package indexes, like PyPI
+    default: "true"
+
+pipeline:
+  - name: Setup pip credentials
+    runs: |
+      set -euo pipefail
+
+      # If libraries token exists in melange's cache, use that.
+      # Fetch the token with: make lib-token
+      if [ -f /var/cache/melange/.libraries_token.txt ]; then
+        cgtoken=$(cat /var/cache/melange/.libraries_token.txt)
+      else
+        idtoken=$(curl --fail-with-body -sSL -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=issuer.enforce.dev)
+        cgtoken=$(curl --fail-with-body -sSL -H "Authorization: Bearer ${idtoken}" 'https://issuer.enforce.dev/sts/exchange?aud=libraries.cgr.dev&scope=${{inputs.group}}&identity=${{inputs.identity}}' | jq -r '.token')
+      fi
+
+      echo "Removing pip config, if exists..."
+      rm -rf "$HOME/.config/pip" && mkdir -p "$HOME/.config/pip"
```

Comment:
Should we error here instead of continuing? That would make debugging easier, in case some package was relying on an existing configuration here.